Export data as CSV and JSON

SelmaGuedidi opened the pull request:
Closes #5

idris commented:
@SelmaGuedidi can you please remove the `output/yaml` directories?  Those are old and not tied to the actual data.  I think it was just a sample format from before that's no longer relevant.

idris commented on .github/workflows/check_yaml_consistency.yml:
```diff
@@ -0,0 +1,21 @@
+        run: pip install -r requirements.txt
+      - name: Run export and check consistency
+        run: |
+          python export.py
+          git diff --exit-code
```
Nice!  Thanks for adding this check

idris commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,9 @@
+        name: Export YAML to CSV and JSON
+        language: python
+        entry: python export.py
+        additional_dependencies: ['pyaml']
```
FYI: `requirements.txt` uses `pyyaml`, not `pyaml`

idris reviewed (changes requested):
Thanks for the PR!  Looks mostly good!  But unfortunately you used the wrong source data.  Sorry for the confusion!

SelmaGuedidi commented:
@idris I made the changes , hope it's fixed :)

- Removed data in output/yaml
- Updated data retrieval to follow schemas in data/
- CSV output generates separate files for companies and brands
- JSON output consolidated into a single file with "brands" and "companies" keys
- Updated requirements.txt to use pyyaml instead of pyaml

idris commented on export.py:
```diff
@@ -0,0 +1,85 @@
+    return data
+
+def clean_value(value):
+    if isinstance(value, list):
+        return ', '.join(map(str, value))
```
The format of dictionaries in the CSV was weird, so I figure JSON is a good format, and it works well for both lists and dicts:
```suggestion
    if isinstance(value, list) or isinstance(value, dict):
        return json.dumps(value)
```

SelmaGuedidi replied:
@idris 

```
   if isinstance(value, list):
        return ', '.join(map(str, value))
```
    
brands.csv:
    
<img width="725" alt="Screenshot 2024-01-18 at 7 15 18 PM" src="https://github.com/TechForPalestine/boycott-israeli-consumer-goods-dataset/assets/79056754/56d97b86-6a2c-4334-8700-4a1dfa0570ed">


```
   if isinstance(value, list) or isinstance(value, dict):
        return json.dumps(value)
```

 brands.csv:

<img width="772" alt="Screenshot 2024-01-18 at 7 19 48 PM" src="https://github.com/TechForPalestine/boycott-israeli-consumer-goods-dataset/assets/79056754/5bb10c6c-0a7a-439a-a411-c18d2945bf82">


I used `', '.join(map(str, value))` to remove `[]` and `""` in the csv files.
and I think dict format does not exist here because `isinstance(value, dict)` is always `false`

idris replied:
Hm, I think you're right that the `', '.join(...)` version is better for lists.

If you look at the sabra row, there is a dict for the `stakeholders` column, which shows up like this:
`"{'id': 'pepsico', 'type': 'owner', 'percent': 50}, {'id': 'strauss-group', 'type': 'owner', 'percent': 50}"`

We're not really sure how the `stakeholders` bit will pan out, so I'm just going to leave your join as-is, and remove `stakeholders` from the CSV export for now.

idris replied:
oooh the plot thickens.. it's an array of objects.  Even more reason to ignore that for now.

idris commented:
Thanks @SelmaGuedidi !  I added one suggestion above, but let me know if you disagree.

@THM222 I think we can handle both timestamps in a separate issue.  I created that here: https://github.com/TechForPalestine/boycott-israeli-consumer-goods-dataset/issues/10
